Dungeons.lua

```diff
@@ -0,0 +1,842 @@
+		
+	end
+
+	-- Now check if the name changed (whether it's SM or RFC or whatever)
+	-- This should normally not happen, as once we're outside, the current dungeon is queued
+	if Hardcore_Character.dt.current.name ~= name then
+		-- Change to the new dungeon, but we store only if we spent enough time
+		Hardcore:Debug( "Left dungeon " .. Hardcore_Character.dt.current.name .. " for dungeon " .. name )
+		DungeonTrackerLogRun( Hardcore_Character.dt.current )
+		Hardcore_Character.dt.current = {}
+	end
+
+	return name
+
+end
+
+-- DungeonTrackerReceivePulse( data, sender )
+--
+-- Receives a group pulse, storing the time in the message and the sender in the associated pending run
+
+function DungeonTrackerReceivePulse( data, sender )
+
+	local shortName
+	local ping_time
+	local dungeon_name
+	local run_name
+
+	shortName, ping_time, dungeon_name = string.split(COMM_FIELD_DELIM, data)
+	ping_time = tonumber( ping_time )
+	Hardcore:Debug( "Received dungeon group pulse from " .. sender .. ", data = " .. shortName .. ", " .. ping_time .. ", " .. dungeon_name ) 
+	
+	-- Check for errors, dt might not be set right now (if it just got reset for some weird reason)
+	if  (Hardcore_Character.dt == nil) or 
+		(not next( Hardcore_Character.dt )) or 
+		(not next( Hardcore_Character.dt.pending )) then
+		return
+	end
+	
+	-- Update the latest ping time in the idle runs only (no need to do it in current run)
+	for i, v in pairs( Hardcore_Character.dt.pending ) do
+		-- If we receive a pulse from "Scarlet Monastery" (without wing), then we have no choice but
+		-- to store that pulse in all idle SM runs. So then we don't care about the wing of the pending run.
+		if dungeon_name == "Scarlet Monastery" then
+			run_name = string.sub( v.name, 1, 17 )		-- This also cuts "The Temple of Atal'Hakkar" to "The Temple of Ata", but that's okay
+		else
+			run_name = v.name
+		end
+		
+		-- If this is the run from which the ping originated, and the ping time is later than we already have, store it
+		if run_name == dungeon_name then
+			if ping_time > v.last_pulse then
+				v.last_pulse = ping_time
+			end
+			
+			-- Add the ping sender to the party members, if not already there
+			if string.find( v.party, shortName ) == nil then
+				v.party = v.party .. "," .. shortName
+			end			
+		end
+	end
+end
+
+-- DungeonTrackerSendPulse( now )
+--
+-- Sends a group pulse, if the time out is expired
+
+function DungeonTrackerSendPulse( now )
+
+	-- Don't send too many pulses, one every 30 seconds is enough
+	if (Hardcore_Character.dt.sent_pulse ~= nil) and 
+	   (now - Hardcore_Character.dt.sent_pulse < DT_GROUP_PULSE) then
+		return
+	end
+	Hardcore_Character.dt.sent_pulse = now
+
+	-- Send my own info to the party (=name + server time + dungeon)
+	if( CTL ) then
+		local name, serverName = UnitFullName("player")
+		local commMessage = COMM_COMMANDS[15] .. COMM_COMMAND_DELIM .. name .. COMM_FIELD_DELIM .. now .. COMM_FIELD_DELIM .. Hardcore_Character.dt.current.name
```

Comment:
Is this working after the refactor? `COMM_COMMANDS` is local to `hardcore.lua`

Reply:
Weird, I am sure it got called, but didn't throw a script error. But fixed, anyway.

Reply:
Wait, CTL doesn't exist in Dungeons.lua, so the if statement failed

Reply:
Alright, needed to copy over a bunch of COMM_ defines from Hardcore.lua (that weren't exported). Works now.